### PR TITLE
Pass runtime configs & variables to BeamSqlSeekableTable

### LINKED ISSUE #28145: [Feature Request]: Access PipelineOptions in BeamSqlSeekableTable
### What would you like to happen?

I'm creating a BeamSqlSeekableTable class, and want to access PipelineOptions.
For example, a HadoopDataSeekableTable needs loading data from Hadoop, and our Hadoop cluster needs keberos auth. 
So, my question is that is it possible to access pipeline options to complete the auth?

I know that I can pass auth parameters to table properties, however, it will be convenient to get parameters from PipelineOptions.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by gabrywu**
fixes #28145

### CODE REVIEW

**Comment by liferoad** [comment]
R: @Abacn

**Comment by liferoad** [comment]
R: @bvolpato

**Comment by gabrywu** [comment]
precommit

**Comment by bvolpato** [comment]
Thanks! No major concerns from me -- just one note is that relying on PipelineOptions will allow configuring behavior for the entire job.

Sometimes it's better to use a specific configuration class / instance, so you could likely have multiple JOINs in the same job and each handle different things.

**Review by bvolpato — APPROVED** [review]

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSqlSeekableTable.java [hunk 1]
```diff
@@ -29,6 +31,12 @@ public interface BeamSqlSeekableTable extends Serializable {
   /** prepare the instance. */
   default void setUp() {};
 
+  default void startBundle(
+      DoFn<Row, Row>.StartBundleContext context, PipelineOptions pipelineOptions) {};
```
Comment (Abacn) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSqlSeekableTable.java:35]: minor - unnecessary semicolon

same as existing setUp() and tearDown()

**Review by Abacn — COMMENTED** [review]
LGTM. It's totally reasonable to add the method as they corresponding the dofn lifecycle methods. Probably we should add them at the beginning.

- One thing to note when implementing youe own BeamSqlSeekableTable. startBundle can be called many times, so if initializing a connection in it, it's good to use a connection pool otherwise there could be too much connection - a common issue seen in Beam IO implementation

also, its nice to have a test checking dofn lifecycle is effective with BeamSqlSeekableTable methods.

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSqlSeekableTable.java (hunk 1 above)
Comment (gabrywu) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSqlSeekableTable.java]: remvoed

**Comment by gabrywu** [comment]
> LGTM. It's totally reasonable to add the method as they corresponding the dofn lifecycle methods. Probably we should add them at the beginning.
> 
> * One thing to note when implementing youe own BeamSqlSeekableTable. startBundle can be called many times, so if initializing a connection in it, it's good to use a connection pool otherwise there could be too much connection - a common issue seen in Beam IO implementation
> 
> also, its nice to have a test checking dofn lifecycle is effective with BeamSqlSeekableTable methods.

thanks very much. I will add initialized flag to startBundle in our own BeamSqlSeekableTable to avoid too much connection.

**Pull request merged**